### ISSUE TITLE
Box Search Manager has a design flaw

### ISSUE THREAD

**Issue by jdbaker** [box/box-windows-sdk-v2#510]
To demonstrate this problem, try receiving the mdfilters as a json string instead of creating them as an anonymous object. Now turn the mdfilters json into an anonymous object using... what exactly? If newtonsoft then you have to create an instance of an object that has all the members you need. But which members do you need? Why, just look at the json of course!

Now that you've somehow overcome that arduous task of serializing json into an anonymous type, let the sdk go ahead and turn that into json again for you. Seems like a good plan. Thanks Box!

But maybe there could be a way to receive what is essentially a dynamic object without having to make a strongly typed one just to turn it back into a dynamic object? Seems like a json string would be ideal.

**Comment by mattwiller** [comment]
@jdbaker You're right — this interface doesn't seem ideal.  I'm not sure that we want to go all the way to taking a JSON string as the parameter value either, but what if we extended the `BoxMetadataFilterRequest` class to include helper methods for constructing the correct filters?  Would something like the following imaginary interface work for you?

```c#
using Box.V2.Models.Request;

var mdfilter = new BoxMetadataFilterRequest(scope="enterprise", templateKey="myMetadata");
mdfilter.equals("/field1", "value1");
mdfilter.lessThan("/field2", 100);

var mdfilters = new List<BoxMetadataFilterRequest>();
mdfilters.Add(mdfilter);

var results = await client.SearchManager.SearchAsync(mdfilters=mdfilters);
```
